lib/rom/relation.rb

```diff
@@ -42,6 +42,10 @@ def initialize(dataset, registry = {})
       @__registry__ = registry
     end
 
+    def self.register_as(value = ROM::ClassMacros::Undefined)
+      super || self.base_name
+    end
+
```

Comment:
Yep this is what we want. This reminds me - could you also move `Undefined` to `ROM::Undefined`? We should use it in many places :)